src/OpenIddict.Client.WebIntegration/OpenIddictClientWebIntegrationProviders.xml

```diff
@@ -738,6 +738,40 @@
     </Environment>
   </Provider>
 
+  <!--
+                                             ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
+                                             ██ ██ ██ ██ █ ▄▄▀██ ███ ██ ▄▄▄█▄ ▄█
+                                             ██ ▄▄ ██ ██ █ ▀▀ ██ █ █ ██ ▄▄▄██ ██
+                                             ██ ██ ██▄▀▀▄█ ██ ██▄▀▄▀▄██ ▀▀▀█▀ ▀█
+                                             ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
+  -->
+
+  <Provider Name="Huawei" Id="95a620d2-a362-47b0-b372-26fa9d41d20e"
+            Documentation="https://developer.huawei.com/consumer/cn/doc/HMSCore-Guides/open-platform-oauth-0000001053629189">
+    <Environment Issuer="https://accounts.huawei.com/"
+                 ConfigurationEndpoint="https://oauth-login.cloud.huawei.com/.well-known/openid-configuration">
```

Comment:
That discovery document doesn't contain a `userinfo_endpoint` address, which means the custom logic you added is not effective since OpenIddict won't send a userinfo request. We could amend the dynamic configuration, but I see in the documentation that the API exposing `GOpen.User.getInfo` seems to be deprecated:

![image](https://github.com/openiddict/openiddict-core/assets/6998306/25fd8af8-12cc-4ca7-9b59-f70217191142)

https://developer.huawei.com/consumer/cn/doc/HMS-2-References/hmssdk_huaweiid_getacountinfo

Does the identity token contain any useful information? If so, we should just ignore the userinfo part.

Reply:
It's a deprecated version of docs.